src/blocksparsearray/blocksparsearray.jl

```diff
@@ -1,168 +1,225 @@
-using BlockArrays: BlockArrays, Block, BlockedUnitRange, blockedrange, blocklength
+using BlockArrays:
+  BlockArrays,
+  Block,
+  BlockedUnitRange,
+  UndefBlocksInitializer,
+  blockedrange,
+  blocklength,
+  undef_blocks
 using DerivableInterfaces: @interface
 using Dictionaries: Dictionary
 using SparseArraysBase: SparseArrayDOK
 
-# TODO: Delete this.
-## using BlockArrays: blocks
+"""
+    SparseArrayDOK{T}(undef_blocks, axes)
+    SparseArrayDOK{T,N}(undef_blocks, axes)
+
+Construct the block structure of an undefined BlockSparseArray that will have
+blocked axes `axes`.
+
+Note that `undef_blocks` is defined in
+[BlockArrays.jl](https://juliaarrays.github.io/BlockArrays.jl/stable/lib/public/#BlockArrays.undef_blocks)
```

Comment:
We could consider re-exporting that too if it turns out we are using this often

Reply:
Right, that touches on a broader question I've been grappling with about how much we want to re-export things. So far my strategy with these new packages is to be very conservative about what we export, including not re-exporting anything and expecting users to load the packages where the objects originated themselves, but once the dust settles a bit and we start using these packages more I'm open to some targeted re-exports.